pkg/k8s/job.go

```diff
@@ -49,6 +49,23 @@ func NewPerfJob(objectMeta metav1.ObjectMeta, app string, imageSpec perfv1alpha1
 		}
 	}
 
+	if podConfig.PodScheduling.Affinity == nil {
```

Comment:
I suppose this fix is somehow related to the Kafka enhancement. Could you please let me know why is this required?

Reply:
I believe this was to prevent an API error that disallows you to partially define the affinities. This pre-populates the sub affinities with empty objects. It allowed me to check the PodAffinity like so:

```
	if job.Spec.Template.Spec.Affinity.PodAffinity == nil {
```